lists-service/cicd/build/pipeline/pipeline.yaml

```diff
@@ -0,0 +1,333 @@
+  pBootstrapStackName:
+    Type: String
+    Description: the name of the bootstrap stack
+  pBucketsStackName:
+    Type: String
+    Description: the name of the bucket stack
+  pCleanBranch:
+    Type: String
+    Description: The clean branch, can be used in resource names
+  pEnvironment:
+    Type: String
+    Description: The AWS environment this belongs to
+  pGitHubBranch:
+    Type: String
+    Description: GitHub branch we're deploying from
+  pGitHubOwner:
+    Type: String
+    Description: GitHub owner
+  pGitHubRepositoryName:
+    Type: String
+    Description: GitHub repository name.
+  pPipelineFingerprint:
+    Type: String
+    Description: The fingerprint of the pipeline, used to determine if it's been modified
+  pProductComponent:
+    Type: String
+    Description: The name of the product component
+  pProductName:
+    Type: String
+    Description: The name of the product
+  pRestartExecutionOnUpdate:
+    Type: String
+    Description: Restart that pipeline if it's been updated
+    AllowedValues:
+         - true 
+         - false
+
+Conditions:
+
+  IsDev: !Equals
+    - !Ref pEnvironment
+    - development
+  AutoDeploy: !Equals
+    - !Ref pAutoDeploy
+    - true
+
+Resources:
+ 
+  ExportConfig:
+    Type: AWS::CodeBuild::Project
+    Properties:
+      Name: !Sub
+              - ${pProductName}-${pProductComponent}-export-config-${ResourceName}
+              - ResourceName: !If [ IsDev, !Ref pCleanBranch, !Ref pEnvironment ]
+      Description: Load the environment and build the CloudFormation template config file
+      ServiceRole:
+               Fn::ImportValue:
+                 Fn::Sub: ${pBootstrapStackName}-CodeBuildServiceRoleArn
+      Artifacts:
+        Type: CODEPIPELINE
+      Environment:
+        Type: LINUX_CONTAINER
+        ComputeType: BUILD_GENERAL1_SMALL
+        Image: aws/codebuild/standard:7.0
+        EnvironmentVariables:
+          - Name: ARTIFACTS_BUCKET
+            Value:
+              Fn::ImportValue:
+                Fn::Sub: ${pBucketsStackName}-${AWS::Region}-CodePipelineArtifactBucketName
+      Source:
+        Type: CODEPIPELINE
+        BuildSpec: !Sub lists-service/cicd/${pProductComponent}/pipeline/export_config_buildspec.yaml
+      TimeoutInMinutes: 5
+
+  BuildLists:
+    Type: AWS::CodeBuild::Project
+    Properties:
+      Name: !Sub
+        - ${pProductName}-${pProductComponent}-build-lists-${ResourceName}
+        - ResourceName: !If [ IsDev, !Ref pCleanBranch, !Ref pEnvironment ]
+      Description: Build the lists project
+      ServiceRole:
+               Fn::ImportValue:
+                 Fn::Sub: ${pBootstrapStackName}-CodeBuildServiceRoleArn
+      Artifacts:
+        Type: CODEPIPELINE
+      Environment:
+        Type: LINUX_CONTAINER
+        ComputeType: BUILD_GENERAL1_SMALL
+        Image: aws/codebuild/standard:7.0
+        EnvironmentVariables:
+          - Name: ENVIRONMENT
+            Value: !Ref pEnvironment
```

Comment:
It's more of a style thing but I've been adding environment variables only to pipeline action and not to the codeBuild project. They work in both places but it gets confusing when they are split between the two, you have to look in and maintain two separate places, in this case it could go in the action on line 243

Reply:
How do we pass `EnvironmentVariables` to code build that are stored in secrets manager when defining in the pipeline action?
You can natively pass using the format:
```
EnvironmentVariables:
  - Name: MONGODB_PASSWORD
    Type: SECRETS_MANAGER
    Value: !Sub
        - 'lists-${ResourceName}:db-password'
        - ResourceName: !If [ IsDev, !Ref pCleanBranch, !Ref pEnvironment ]
```
I assume that this read updated secret values

Reply:
oh cool, I didnt know about that one. You can get secrets natively in the codebuild buildspec using 

```
env:
  shell: bash
  git-credential-helper: yes
  variables:
    DEBIAN_FRONTEND: noninteractive
  secrets-manager:
    USERDETAILS_OIDC_SECRET: "$PRODUCT_NAME-$RESOURCE_NAME:userdetails_oidc_secret"
    USERDETAILS_ENDPOINT_PASSWORD: "$PRODUCT_NAME-$RESOURCE_NAME:userdetails_endpoint_password"
```
( just the secrets-manager bit, included the rest for context )

Reply:
In this repo: https://github.com/AtlasOfLivingAustralia/species-lists/blob/ci-cd/lists-service/cicd/deploy/pipeline/deploy_lists_buildspec.yaml#L11